src/charm.py

```diff
@@ -124,59 +126,99 @@ def __init__(self, *args):
     def ready_to_configure(self) -> bool:
         """Returns whether all preconditions are met to proceed with configuration."""
         if not self._container.can_connect():
-            self.unit.status = WaitingStatus("Waiting for container to be ready")
             return False
         for relation in [DATABASE_RELATION_NAME, "certificates"]:
             if not self._relation_created(relation):
-                self.unit.status = BlockedStatus(f"Waiting for {relation} relation to be created")
                 return False
         if not self._database_is_available():
-            self.unit.status = WaitingStatus("Waiting for the database to be available")
             return False
         if not self._get_database_uri():
-            self.unit.status = WaitingStatus("Waiting for database URI")
             return False
         if not self._container.exists(path=BASE_CONFIG_PATH) or not self._container.exists(
             path=CERTS_DIR_PATH
         ):
-            self.unit.status = WaitingStatus("Waiting for storage to be attached")
             return False
         if not _get_pod_ip():
-            self.unit.status = WaitingStatus("Waiting for pod IP address to be available")
             return False
         return True
 
+    def _on_collect_unit_status(self, event: CollectStatusEvent):  # noqa C901
+        """Check the unit status and set to Unit when CollectStatusEvent is fired.
+
+        Args:
+            event: CollectStatusEvent
+        """
+        if not self.unit.is_leader():
+            # NOTE: In cases where leader status is lost before the charm is
+            # finished processing all teardown events, this prevents teardown
+            # event code from running. Luckily, for this charm, none of the
+            # teardown code is necessary to perform if we're removing the
+            # charm.
+            event.add_status(BlockedStatus("Scaling is not implemented for this charm"))
```

Comment:
WDYT about logging after updating the status ?

Reply:
I added logs after setting every status.